test/extended/util/cloud/cloud.go

```diff
@@ -0,0 +1,75 @@
+package cloud
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+)
+
+// LoadConfig uses the cluster to setup the cloud provider config.
+func LoadConfig() (string, *e2e.CloudConfig, error) {
+	coreClient, err := e2e.LoadClientset()
+	if err != nil {
+		return "", nil, err
+	}
+	clientConfig, err := e2e.LoadConfig()
+	if err != nil {
+		return "", nil, err
+	}
+	client := configclient.NewForConfigOrDie(clientConfig)
+
+	infra, err := client.ConfigV1().Infrastructures().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return "", nil, err
+	}
+	p := infra.Status.PlatformStatus
+	if p == nil {
+		return "", nil, fmt.Errorf("status.platformStatus must be set")
+	}
+	if p.Type == configv1.NonePlatformType {
+		return "", nil, nil
+	}
+
```

Comment:
IMO we should restructure as 
```

## all the platform agnostic settings like zones, zone, multimaster etc.

## switch and return directly to avoid these variables `projectID` `region`

```

Reply:
ok